packages/sync/src/modules/class-full-sync-immediately.php

```diff
@@ -0,0 +1,405 @@
+		$this->send_action( 'jetpack_full_sync_start', array( $full_sync_config, $range ) );
+
+		return true;
+	}
+
+	/**
+	 * Whether full sync has started.
+	 *
+	 * @access public
+	 *
+	 * @return boolean
+	 */
+	public function is_started() {
+		return ! ! $this->get_status()['started'];
+	}
+
+	/**
+	 * Retrieve the status of the current full sync.
+	 *
+	 * @access public
+	 *
+	 * @return array Full sync status.
+	 */
+	public function get_status() {
+		$default = array(
+			'started'  => false,
+			'finished' => false,
+			'progress' => array(),
+			'config'   => array(),
+		);
+
+		return wp_parse_args( \Jetpack_Options::get_raw_option( self::STATUS_OPTION ), $default );
+	}
+
+	/**
+	 * Whether full sync has finished.
+	 *
+	 * @access public
+	 *
+	 * @return boolean
+	 */
+	public function is_finished() {
+		return ! ! $this->get_status()['finished'];
+	}
+
+	/**
+	 * Clear all the full sync data.
+	 *
+	 * @access public
+	 */
+	public function reset_data() {
+		$this->clear_status();
+		( new Lock() )->remove( self::LOCK_NAME );
+	}
+
+	/**
+	 * Clear all the full sync status options.
+	 *
+	 * @access public
+	 */
+	public function clear_status() {
+		\Jetpack_Options::delete_raw_option( self::STATUS_OPTION );
+	}
+
+	/**
+	 * Updates the status of the current full sync.
+	 *
+	 * @access public
+	 *
+	 * @param array $values New values to set.
+	 *
+	 * @return bool True if success.
+	 */
+	public function update_status( $values ) {
+		return $this->set_status( wp_parse_args( $values, $this->get_status() ) );
+	}
+
+	/**
+	 * Retrieve the status of the current full sync.
+	 *
+	 * @param array $values New values to set.
+	 *
+	 * @access public
+	 *
+	 * @return boolean Full sync status.
+	 */
+	public function set_status( $values ) {
+		return \Jetpack_Options::update_raw_option( self::STATUS_OPTION, $values );
+	}
+
+	/**
+	 * Given an initial Full Sync configuration set the initial status.
+	 *
+	 * @param array $full_sync_config Full sync configuration.
+	 *
+	 * @return array Initial Sent status.
+	 */
+	public function get_initial_progress( $full_sync_config ) {
+		// Set default configuration, calculate totals, and save configuration if totals > 0.
+		$status = array();
+		foreach ( $full_sync_config as $name => $config ) {
+			$module          = Modules::get_module( $name );
+			$status[ $name ] = array(
+				'total'    => $module->total( $config ),   // Total.
+				'sent'     => 0,             // Sent.
+				'finished' => false,
+			);
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Get the range for content (posts and comments) to sync.
+	 *
+	 * @access private
+	 *
+	 * @return array Array of range (min ID, max ID, total items) for all content types.
+	 */
+	private function get_content_range() {
+		$range  = array();
+		$config = $this->get_status()['config'];
+		// Only when we are sending the whole range do we want to send also the range.
+		if ( true === isset( $config['posts'] ) && $config['posts'] ) {
```

Comment:
Seems like we're duplicating a bunch of code from the original full sync module. Should we abstract it?

Reply:
we will kill the original once we bulletproof this one, we want this file as close as we can from it's future version